src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrBackflower.cs

```diff
@@ -256,22 +263,22 @@ internal class VmrBackFlower : VmrCodeFlower, IVmrBackFlower
 
         _logger.LogInformation("New branch {branch} with flown code is ready in {repoDir}", branchName, targetRepo);
 
-        return branchName;
+        return true;
     }
 
-    protected override async Task<string?> OppositeDirectionFlowAsync(
+    protected override async Task<bool> OppositeDirectionFlowAsync(
         SourceMapping mapping,
         Codeflow lastFlow,
         Codeflow currentFlow,
         ILocalGitRepo targetRepo,
         Build? build,
+        string branchName,
         bool discardPatches,
         CancellationToken cancellationToken)
     {
         await targetRepo.CheckoutAsync(lastFlow.SourceSha);
 
-        var branchName = currentFlow.GetBranchName();
-        var patchName = _vmrInfo.TmpPath / $"{branchName.Replace('/', '-')}.patch";
+        var patchName = _vmrInfo.TmpPath / $"{mapping.Name}-{Commit.GetShortSha(lastFlow.VmrSha)}-{Commit.GetShortSha(currentFlow.TargetSha)}.patch";
```

Comment:
Should these patch names be consistent between the different flows? I see a similar name above. 
If they are, maybe its worth having them resolved in a separate one-liner

Reply:
They don't need to be consistent, it's more for humans really. They must differ between different flows though.